autosklearn/data/validation.py

```diff
@@ -0,0 +1,297 @@
+        y: Union[pd.DataFrame, np.ndarray],
+    ) -> np.ndarray:
+        """
+        Wrapper around sklearn check_array. Translates a pandas
+        Dataframe to a valid input for sklearn.
+        """
+        # Support also Pandas object in the targets
+        if hasattr(y, "iloc"):
+            y = self._validate_dataframe_target(y)
+
+        y = np.atleast_1d(y)
+        if y.ndim == 2 and y.shape[1] == 1:
+            warnings.warn("A column-vector y was passed when a 1d array was"
+                          " expected. Will change shape via np.ravel().",
+                          sklearn.utils.DataConversionWarning, stacklevel=2)
+            y = np.ravel(y)
+
+        # sklearn check array will make sure we have the
+        # correct numerical features for the array
+        # Also, a numpy array will be created
+        y = sklearn.utils.check_array(
+            y,
+            force_all_finite=True,
+            accept_sparse='csr',
+            ensure_2d=False,
+        )
+        return y
+
+    def _validate_dataframe_features(
+        self,
+        X: pd.DataFrame,
+    ) -> Union[pd.DataFrame, np.ndarray]:
+        """
+        Interprets a Pandas
+        Uses .iloc as a safe way to deal with pandas object
+        """
+
+        # Start with the features
+
+        # Register if a column needs encoding
+        enc_columns = []
+
+        # Also, register the feature types for the estimator
+        feature_types = []
+
+        # Make sure each column is a valid type
+        for i, column in enumerate(X.columns):
+            if X[column].dtype.name in self.valid_pd_enc_dtypes:
+                enc_columns.append(i)
+                feature_types.append('categorical')
+            elif not np.issubdtype(X[column].dtype, np.number):
+                if X[column].dtype.name == 'Object':
+                    raise ValueError(
+                        "Input Column {} has invalid type object. "
+                        "Cast it to a numerical value before using it in Auto-Sklearn."
+                        "You can use pandas.to_numeric() to do this".format(
+                            column,
+                        )
+                    )
+                elif pd.core.dtypes.common.is_datetime_or_timedelta_dtype(
+                    X[column].dtype
+                ):
+                    raise ValueError(
+                        "Input Column {} has invalid time-type. "
+                        "Please convert the time information to a categorical value. "
+                        "Usually, time information can be separated into a set of categorical "
+                        "features. For example, hour of the day can be converted to 24 "
+                        " categories.".format(
+                            column,
+                        )
+                    )
+                else:
+                    raise ValueError(
+                        "Input Column {} has unsupported dtype {}. "
```

Comment:
Is string a separate type? If yes, could you please add a separate error message and point to [this sklearn tutorial](https://scikit-learn.org/stable/tutorial/text_analytics/working_with_text_data.html)?

Reply:
According to this [link](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dtypes.html) Strings have dtype object. I will include this tutorial there.